### PR TITLE
[UI Components]: Docs examples improvements

### PR DESCRIPTION
**Pull request by oliverigor**
### Background

Following [this discussion](https://github.com/Shopify/checkout-web/discussions/19342), we will be implementing examples added there to our docs, this will help developers find how can they achieve some of the most used patterns utilizing out libraries.

To merge this I'm waiting for a review here:
  - [x] [Checkout web](https://github.com/Shopify/checkout-web/pull/20080) PR

### Solution

- Added some of the examples, with the idea to iterate over them in the future.

### 🎩

- Adding the list of icons available: https://shopify-dev.docs-examples-2.igor-deoliveiramartins.us.spin.dev/docs/api/checkout-ui-extensions/components/media/icon

- Adding some desired patterns through examples [like this one](https://shopify-dev.docs-examples-2.igor-deoliveiramartins.us.spin.dev/docs/api/checkout-ui-extensions/components/feedback/skeletonimage#examples)

### Checklist

- [x] I have :tophat:'d these changes
- [x] I have updated relevant documentation

### CODE REVIEW

File: packages/checkout-ui-extensions/docs/build-docs.sh [hunk 1]
```diff
@@ -10,7 +10,7 @@ find ./ -name '*.doc*.js' -exec rm -r {} \;
 
 if [ $build_exit -ne 0 ]; then
   echo "** Failed to generate docs"
-  echo "See https://checkout.docs.shopify.io/docs/referencedocs/web/extensions/docs"
```
[the remark this answers is not in the document]
Comment (jamesvidler) [reply packages/checkout-ui-extensions/docs/build-docs.sh:13]: Oh, yes thank you for changing that. The docs were just migrated to vault a couple days ago. 👍

File: packages/checkout-ui-extensions/src/components/shared.ts [hunk 2]
```diff
@@ -335,7 +335,7 @@ export interface SpacingProps {
    *
    * - [`base`, `none`] means blockStart and blockEnd paddings are `base`, inlineStart and inlineEnd paddings are `none`
    *
-   * - [`base`, `none`, `loose`, `tight`] means blockStart padding is `base`, inlineEnd padding is `none`, blockEnd padding is `loose` and blockStart padding is `tight`
+   * - [`base`, `none`, `loose`, `tight`] means blockStart padding is `base`, inlineEnd padding is `none`, blockEnd padding is `loose` and  blockStart padding is `tight`
```
Comment (ncardeli) [inline packages/checkout-ui-extensions/src/components/shared.ts:338]: Nittest of the nits: there's an extra-space before "blockStart"

**Review by ncardeli — APPROVED** [review]
LGTM!

File: packages/checkout-ui-extensions/src/components/shared.ts (hunk 2 above)
Comment (oliverigor) [reply packages/checkout-ui-extensions/src/components/shared.ts:338]: oh, will fix that in checkout web once we push DatePicker.

File: packages/checkout-ui-extensions/src/extension-points/api/standard/index.ts [hunk 3]
```diff
@@ -374,7 +374,7 @@ export interface AppMetafieldEntry {
   metafield: AppMetafield;
 }
 
-export type ApiVersion = '2022-10' | '2023-01' | 'unstable';
```
[the remark this answers is not in the document]
Comment (jamesvidler) [reply packages/checkout-ui-extensions/src/extension-points/api/standard/index.ts:377]: This is ok 👍

**Review by jamesvidler — COMMENTED** [review]
Nice! I love the additional component examples 🔥.

I'm assuming the commits in this PR are also present in `checkout-web`? 

Lastly, do you think we could try to re-base from `checkout-web` to include [this PR](https://github.com/Shopify/checkout-web/pull/20484) when it ships? That would set us up nicely to enable multiple api version support in shopify-dev without having to publish the packages twice.

**Review by jamesvidler — APPROVED** [review]
As discussed, you don't need to wait for this PR https://github.com/Shopify/checkout-web/pull/20484 since you are on a tight deadline. Nice work! Let me know if you'd like me to review any other package related PRs 👍

**Pull request merged**